android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/RegisterScreen.kt

```diff
@@ -160,15 +187,100 @@ fun RegisterScreen(
+        RegisterCardList(
+          registerCardConfig =
+            registerViewModel.retrieveRegisterConfiguration(registerId).registerCard,
+          pagingItems = pagingItems
+        ) { viewComponentEvent ->
+          registerViewModel.onEvent(
+            RegisterEvent.OnViewComponentEvent(viewComponentEvent, navController)
+          )
+        }
+      } else {
+        registerConfiguration.noResults?.let { noResultConfig ->
+          NoRegistersView(
+            modifier = modifier,
+            context = context,
+            noResults = noResultConfig,
+            onClick = {
+              val onClickAction =
+                noResultConfig.actionButton?.actions?.find { it.trigger == ActionTrigger.ON_CLICK }
+              onClickAction?.let { actionConfig ->
+                when (onClickAction.workflow) {
+                  ApplicationWorkflow.LAUNCH_REGISTER -> {
+                    actionConfig.questionnaire?.id?.let { questionnaireId ->
+                      context.launchQuestionnaire<QuestionnaireActivity>(
+                        questionnaireId = questionnaireId
+                      )
+                    }
+                  }
+                  else -> {}
+                }
+              }
+            }
+          )
+        }
       }
     }
   }
 }
+
+@Composable
+fun NoRegistersView(
+  modifier: Modifier = Modifier,
+  context: Context,
+  noResults: NoResultsConfig,
+  onClick: () -> Unit
+) {
+  Column(
+    modifier = modifier.fillMaxSize().padding(16.dp).testTag(NO_REGISTER_VIEW_COLUMN_TEST_TAG),
+    horizontalAlignment = Alignment.CenterHorizontally,
+    verticalArrangement = Arrangement.Center
+  ) {
+    Text(
+      text = noResults.title,
+      fontSize = 16.sp,
+      modifier = modifier.padding(vertical = 8.dp).testTag(NO_REGISTER_VIEW_TITLE_TEST_TAG),
+      fontWeight = FontWeight.Bold
+    )
+    Text(
+      text = noResults.message,
+      modifier =
+        modifier.padding(start = 32.dp, end = 32.dp).testTag(NO_REGISTER_VIEW_MESSAGE_TEST_TAG),
+      textAlign = TextAlign.Center,
+      fontSize = 15.sp,
+      color = Color.Gray
+    )
+    Button(
+      modifier = modifier.padding(vertical = 16.dp).testTag(NO_REGISTER_VIEW_BUTTON_TEST_TAG),
+      onClick = onClick
+    ) {
+      Icon(
+        imageVector = Icons.Filled.Add,
+        contentDescription = null,
+        modifier.padding(end = 8.dp).testTag(NO_REGISTER_VIEW_BUTTON_ICON_TEST_TAG)
+      )
+      Text(
+        text = noResults.actionButton?.display?.uppercase().toString(),
+        modifier.testTag(NO_REGISTER_VIEW_BUTTON_TEXT_TEST_TAG)
+      )
+    }
+  }
+}
+
+@Preview(showBackground = true)
+@ExcludeFromJacocoGeneratedReport
+@Composable
+private fun PreviewNoRegistersView() {
+  NoRegistersView(
+    noResults =
+      NoResultsConfig(
+        title = "Title",
+        message = "This is message",
+        actionButton = NavigationMenuConfig(display = "Button Text", id = "1")
```

Comment:
Please move this to the `strings.xml` file

Reply:
This is just the `Preview` method to preview and verify the composable, throughout our app we've used these hardcoded dummy strings for populating such previews

Reply:
@dubdabasoduba We can resolve this conversation.